### PR TITLE
Naming Patterns added to Architecture

### PR DESCRIPTION
**Pull request by mhfen**
@justinhillsjohnson I've added Naming Pattern guidelines to further help adoption of the new architecture. Please let me know your thoughts.

@uxtronaut Would love your thoughts here as well.

@lukemccomb Do you think these guidelines would be helpful when learning this architecture? You are the developer with the most experience learning them so far.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,3 +196,131 @@ These assertions are widely accepted, however, there is no standard approach to
...
+
+```
+
+#### Sub Components
+
+Use a hyphen before a class member (sub-component) name, prefixed with the base class name. The class member name is camelCased. Sub components should be direct children of their parent base module element.
+
+```css
+
+  .feature-hd {
```
Comment (uxtronaut) [inline README.md]: What do you think about avoiding abbreviations or sub-strings in naming conventions. Personally i would rather see

```css
  .feature-header
  .feature-body
  .feature-footer
```

Than `-hd`, `-bd`, or `-ft`.

File: README.md [hunk 2]
```diff
@@ -196,3 +196,131 @@ These assertions are widely accepted, however, there is no standard approach to
...
+
+```
+
+#### Mixins
+
+Use “mix-” followed by the base class, then the rest of the class extension name. Mixins typically contain only one or two properties and are interchangeable.
+
+```css
+
+  .hdg {
```
Comment (uxtronaut) [inline README.md]: Similar to my previous note, here I would rather just see `.heading` than decipher what `.hdg` meant.

Comment (justinhillsjohnson) [inline README.md]: :+1: To Zach's comments on naming conventions

File: README.md [hunk 3]
```diff
@@ -196,3 +196,131 @@ These assertions are widely accepted, however, there is no standard approach to
...
+    color: #ff0000;
+    font-family: Times, "Times New Roman", serif;
+  }
+
+```
+
+#### Sub Components
+
+Use a hyphen before a class member (sub-component) name, prefixed with the base class name. The class member name is camelCased. Sub components should be direct children of their parent base module element. Prefer verbose naming (`.feature-header`) over abbreviation (`.feature-hd`).
+
```
Comment (mhfen) [inline README.md]: @uxtronaut @justinhillsjohnson I've updated the guidelines with the note above and more verbose examples. Thanks for the input!

**Pull request merged**